cloudflare/resource_cloudflare_record.go

```diff
@@ -374,13 +374,20 @@ func resourceCloudflareRecordCreate(d *schema.ResourceData, meta interface{}) er
 		r, err := client.CreateDNSRecord(context.Background(), newRecord.ZoneID, newRecord)
 		if err != nil {
 			if strings.Contains(err.Error(), "already exist") {
-
 				if d.Get("allow_overwrite").(bool) {
+					var r cloudflare.DNSRecord
 					log.Printf("[DEBUG] Cloudflare Record already exists however we are overwriting it")
 					zone, _ := client.ZoneDetails(context.Background(), d.Get("zone_id").(string))
-					r := cloudflare.DNSRecord{
-						Name: d.Get("name").(string) + "." + zone.Name,
-						Type: d.Get("type").(string),
+					if d.Get("name").(string) == "@" {
```

Comment:
not everyone uses the alias here; if we're adding this, it needs to account for both formats.

Reply:
What will the API accept as a valid input for root domain? `""` or `"."`?

Reply:
It will be the entire zone name (such as `example.com`) from the API response.

Reply:
This should be fixed